### PR TITLE
Feat/#38 Create SfPagination

### PR DESCRIPTION
**Pull request by leomp12**
# Related issue

https://github.com/DivanteLtd/storefront-ui/issues/38

# Scope of work

- Update dependencies;
- Added chevron svg icons to assets folder;
- New `SfPagination` component adapted from [Alpaca Pager](https://github.com/SnowdogApps/alpaca-storybook/tree/develop/components/src/03-modules/pager) component;
- Adapted styles, kept same props from Alpaca's version, added slots `prev` and `next` to enable replacement of default chevron icons;
- Unit tests, stories and md reference for `SfPagination` component;

# Screenshots of visual changes
pagination

### CODE REVIEW

File: src/components/molecules/SfPagination/SfPagination.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+export default {
+  props: {
+    page: {
```
[the remark this answers is not in the document]
Comment (leomp12) [reply src/components/molecules/SfPagination/SfPagination.js]: Yup, I'm also thinking about renaming `limit` to `page-size`...

File: src/components/molecules/SfPagination/README.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+<!-- Describe slots and their purpose -->
+
+- `prev` - Previous page link, contents chevron left icon by default.
+
+- `next` - Next page link, contains chevron right icon by default.
+
+## Emitted events
+
+- `update:page` - Emitted when an pagination link is clicked and
```
[the remark this answers is not in the document]
Comment (leomp12) [reply src/components/molecules/SfPagination/README.md]: At the truth I've just copied it from Alpaca Pager, but I think it's not a good practice (not sure) to use `change` event as it already exists as default for form elements, maybe `current-change`?

[the remark this answers is not in the document]
Comment (filrak) [reply src/components/molecules/SfPagination/README.md]: You're right. Let's leave it like this

**Comment by filrak** [comment]
Ok i reviewed again - maybe a generał slot for current and other pages is a better idea?  Users should be able to have totally custom html for every part but too mamy slota is also not the best idea

Do you think having slots for prev, next and item + css modifier for current is a good API?

Sorry if it's adding you extra work. It's mostly because of my poorly made review first time. If you want I can merge it and adjust by myself

**Comment by filrak** [comment]
We can also just add slot for current and it's still fine

**Comment by leomp12** [comment]
> Ok i reviewed again - maybe a generał slot for current and other pages is a better idea? Users should be able to have totally custom html for every part but too mamy slota is also not the best idea

I thought about it, but I think we must keep the number out of the slot to do that, right? Maybe we can create another CSS component for the number, so it may be hidden if needed, and the slot will be placed after the number, but I'm not sure if it will be usual.

> Do you think having slots for prev, next and item + css modifier for current is a good API?

I think that would be more common to customize the prev and next icons than the page items, so, imho next and prev slots are the most important, but we can remove the modifiers and use `:last-child` and `:first-child` on SCSS.
Anyway, they're not modifiers for the root component, so I think it's not disturbing the API, maybe to keep it a little bit cleaner...

> Sorry if it's adding you extra work. It's mostly because of my poorly made review first time. If you want I can merge it and adjust by myself

No problem at all, discussing here I can be more closer to the project approach, then the next PRs may need less reviews.

**Comment by filrak** [comment]
Putting number in slot is imho a good idea. I was just thinking about having a way to customize the content of it. 
````HTML
      <a
        class="sf-pagination__link"
        @click="setCurrentPage(number)">
        <slot>
             {{ number }}
        </slot>
      </a>
````
It's prty rare use case so we can leave it like this.

Regarding `page-size` don't you think it's redundant? If i understand correctly it's mean to describe limit parameter in request which not in charge of pagination at all. Pagaination is dumb component and smart wrapper should be taking care of such details. WDYT?

**Comment by leomp12** [comment]
Imho will be more rare to replace the numbers on pagination, by putting the slot this way the user can't put an svg icon on pagination items (for example) without removing the numbers, I can't see cases where the user would like to compose the pagination without page numbers, so my suggestion is:

```html
      <a
        class="sf-pagination__link"
        @click="setCurrentPage(number)">
        <span class="sf-pagination__number">{{ number }}</span>
        <slot></slot>
      </a>
```

I think it's not necessary, but to make it completely customizable we can also create more one slot, parent of the number and the default slot.
Or we can deal with your suggestion but using [scoped slot](https://vuejs.org/v2/guide/components-slots.html#Scoped-Slots-with-the-slot-scope-Attribute).

I'm not sure about `page-size` or `limit`, I saw `page-size` in other libraries so decided to use, also thought about `per-page`, do you think `limit` is better? I think `limit` is better if you are associating in request to the APIs or database, but I think it's not so good with only the pagination component in mind.

**Comment by leomp12** [comment]
As discussed on Slack, I've changed the component API, the prop names are quite simple: `value`, `length`, `visible`. Maybe we can use `page`, `visiblePageNumber` and `numberOfPages` to be clear, I'm not sure, wdyt?
Using default slot for page items as suggested, but I'm also binding the `number` on slot.
New slot `points` (I don't think it's necessary at all, but may be useful...).

**Comment by leomp12** [comment]
Actually `visible`  is the "maximum number of visible page items, default is 5" as described on README, `value` is what I suppose you mean as current.
It's almost the same prop names used on Vuetify for the respective component, but they're using `total-visible` instead of `visible`.

**Comment by filrak** [comment]
Yeah, this was what i meant. Sorry I'm in the airport and cannot think straight ^^. What do you think about `total` `visible` and `current` ?

**Comment by leomp12** [comment]
Sounds better :)
Edited..

**Pull request merged**